service/network.py

```diff
@@ -71,13 +72,29 @@ def request(self, url, type, data=None):
 
         # Set up some things for the request
         versionString = "{0} {1} - {2} {3}".format(config.version, config.tag, config.expansionName, config.expansionVersion)
-        headers={"User-Agent" : "pyfa {0} (Python-urllib2)".format(versionString)}
+        headers = {"User-Agent" : "pyfa {0} (Python-urllib2)".format(versionString)}
 
         proxy = NetworkSettings.getInstance().getProxySettings()
         if proxy is not None:
-            proxy = urllib2.ProxyHandler({'https': "{0}:{1}".format(*proxy)})
-            opener = urllib2.build_opener(proxy)
+            # proxy is tuple of (host, port):  (u'192.168.20.1', 3128)
+            # build default proxy handler
+            proxy_handler = urllib2.ProxyHandler({'https': "{0}:{1}".format(proxy[0], proxy[1])})
+            proxy_auth = NetworkSettings.getInstance().getProxyAuthDetails()
+            if proxy_auth is not None:
```

Comment:
Looking at `getProxyAuthDetails()`, this will never return `None`, instead it will always return a tuple of two strings. So why are we checking if it's `None` here? Doesn't this make a bit more sense:

```
proxy_auth = NetworkSettings.getInstance().getProxyAuthDetails()
if proxy_auth is not None:
    proxy_handler = urllib2.ProxyHandler({'https':  '{0}:{1}@{2}:{3}'.format(proxy_auth[0], proxy_auth[1], proxy[0], proxy[1])})
else:
    proxy_handler = urllib2.ProxyHandler({'https': "{0}:{1}".format(proxy[0], proxy[1])})
```

And then just make the `getProxyAuthDetails()` return None if it doesn't have both username and password?